### ISSUE TITLE
Add unsupervised learning and synthetic target suggestions when no obvious ML target exists

### ISSUE THREAD

**Issue by Mrassimo** [Mrassimo/datapilot#22]
## Feature Request

Section 6 (Predictive Modeling) currently scores 7/10 in usefulness. The main issue is that it gives up when no obvious target variable is found, missing opportunities for unsupervised learning and synthetic target creation.

## Current State

Section 6 provides:
- Ethical AI considerations (excellent)
- Bias detection in sensitive attributes
- Generic workflow steps
- But: "0 suitable modeling tasks identified" when no target variable exists

### 1. Unsupervised Learning Recommendations

When no target variable is found, suggest unsupervised approaches:

```markdown
**6.2.A Unsupervised Learning Opportunities**

Since no target variable was identified, DataPilot recommends:

**Customer Segmentation (K-Means Clustering)**
- Features: Encode Country, City, Company type
- Optimal K: 5-7 clusters (based on elbow method)
- Use case: Marketing campaign targeting
- Implementation: 
  ```python
  from sklearn.cluster import KMeans
  from sklearn.preprocessing import StandardScaler
  
  # Auto-generated code based on your data
  features = ['country_encoded', 'city_encoded', 'company_size']
  kmeans = KMeans(n_clusters=6, random_state=42)
  ```

**Anomaly Detection**
- Method: Isolation Forest on customer attributes
- Identifies: Unusual customer profiles
- Business value: Fraud detection, data quality
- ~2% of records flagged as anomalies
```

### 2. Synthetic Target Variable Creation

Suggest creating meaningful targets from existing data:

```markdown
**6.2.B Synthetic Target Variables**

DataPilot can create these target variables from your data:

1. **Customer Lifetime (Survival Analysis)**
   - Target: Days since subscription
   - Type: Regression/Survival
   - Features: All customer attributes
   - Missing dates: Treat as censored data

2. **Email Domain Quality**
   - Target: Corporate (1) vs Personal (0) email
   - Type: Binary classification  
   - Created from: Email domain analysis
   - Training data: 66% corporate domains detected

3. **Geographic Market Tiers**
   - Target: High/Medium/Low value markets
   - Type: Multi-class classification
   - Based on: Customer density by country
   - Enables: Market prioritization models

4. **Data Completeness Score**
   - Target: % of non-null important fields
   - Type: Regression (0-100%)
   - Use case: Predict data quality by source
```

### 3. AutoML Integration Recommendations

Provide specific AutoML tool recommendations:

```markdown
**6.2.C AutoML Platform Recommendations**

Based on your data characteristics:

**Recommended: H2O AutoML**
- Best for: Mixed data types with high cardinality
- Handles: Automatic encoding of categoricals
- Code:
  ```python
  import h2o
  from h2o.automl import H2OAutoML
  
  # DataPilot-optimized settings for your data
  aml = H2OAutoML(
      max_models=20,
      seed=42,
      exclude_algos=['DeepLearning'],  # High cardinality issues
      max_runtime_secs=3600
  )
  ```

**Alternative: AutoGluon**
- Strength: Handles text features well
- Ideal for: Company name analysis
- Includes: Neural network options
```

### 4. Feature Engineering Recipes

Provide specific feature engineering code:

```markdown
**6.2.D Feature Engineering Cookbook**

DataPilot generated these feature engineering recipes:

**Temporal Features from Subscription Date:**
```python
# Auto-generated based on date range analysis
df['days_since_subscription'] = (pd.Timestamp.now() - df['subscription_date']).dt.days
df['subscription_quarter'] = df['subscription_date'].dt.quarter
df['subscription_dayofweek'] = df['subscription_date'].dt.dayofweek
df['is_weekend_signup'] = df['subscription_dayofweek'].isin([5, 6])
```

**Geographic Enrichment:**
```python
# Country-level features
country_stats = df.groupby('country').agg({
    'customer_id': 'count',
    'company': 'nunique'
}).rename(columns={'customer_id': 'country_customers', 
                   'company': 'country_companies'})
df = df.merge(country_stats, on='country', how='left')
```

**Text-Based Features:**
```python
# Extract patterns from high-cardinality text
df['email_domain'] = df['email'].str.split('@').str[1]
df['is_corporate_email'] = ~df['email_domain'].isin(['gmail.com', 'yahoo.com', 'hotmail.com'])
df['name_length'] = df['first_name'].str.len() + df['last_name'].str.len()
```
```

### 5. Model Deployment Readiness

Add deployment considerations:

```markdown
**6.7 Deployment Readiness Assessment**

**Data Pipeline Requirements:**
- Real-time preprocessing needed for 12 features
- Encoding dictionaries for 9 categorical variables
- Memory requirement: ~500MB for lookup tables

**Monitoring Recommendations:**
- Track country distribution drift (baseline: Congo 0.84%)
- Monitor email domain quality (baseline: 66.6% valid)
- Alert on >10% change in cardinality

**API Schema:**
```json
{
  "input_schema": {
    "customer_id": "string",
    "first_name": "string", 
    "last_name": "string",
    "company": "string",
    "country": "string",
    // ... all fields
  },
  "preprocessing_steps": [
    "validate_email",
    "encode_country",
    "extract_features"
  ],
  "output_schema": {
    "prediction": "float",
    "confidence": "float",
    "segment": "string"
  }
}
```
```

## Implementation Details

1. Add `--suggest-targets` flag to generate synthetic targets
2. Include unsupervised learning in default analysis
3. Generate actual feature engineering code
4. Provide deployment templates
5. Calculate feasibility scores for each approach

## Benefits

- Never returns "0 modeling tasks" - always finds opportunities
- Practical code generation reduces implementation time
- Bridges gap between analysis and production
- Enables ML on any dataset, not just those with labels
- Deployment guidance prevents production surprises

## Example Command

```bash
datapilot modeling data.csv --suggest-targets --generate-features
# Output includes: unsupervised options, synthetic targets, feature engineering code
```
